packages/cascara/src/components/Table/TableBase.js

```diff
@@ -193,21 +183,22 @@ const TableBase = ({
       return data;
     }
 
-    const sortData = sortWith([
-      (sortState?.sortOrder === SORT_ORDER.ASCENDING ? ascend : descend)(
-        // DEV-18614-15: make sorting case insensitive
-        compose(
-          ifElse(is(String), toLower, identity),
-          prop(sortState?.sortAttribute)
-        )
-      ),
-    ]);
+    // find module type
+    const moduleType = display.find(
+      (item) => item.label === sortState?.sortAttribute
+    );
 
-    const sortedData = sortData(data);
+    const sortedData = sortDataType(
+      data,
+      sortState?.sortAttribute,
+      sortState?.sortOrder,
+      moduleType?.module
```

Comment:
If `sortState` or `moduleType` are not present (null), will `sortDataType` function properly?

Reply:
Yes, if those properties don't exist, the sorting won't have any effect.